src/Microsoft.Extensions.FileProviders.Physical/PollingWildCardChangeToken.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.FileProviders.Physical.Internal;
+using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Physical
+{
+    public class PollingWildCardChangeToken : IChangeToken
+    {
+        public static readonly TimeSpan PollingInterval = TimeSpan.FromSeconds(4);
+        private const int DefaultBufferSize = 256;
+        private readonly object _enumerationLock = new object();
+        private readonly DirectoryInfoBase _directoryInfo;
+        private readonly Matcher _matcher;
+        private bool _changed;
+        private DateTime _lastScanTimeUtc;
+        private byte[] _fileNameBuffer;
+        private byte[] _previousHash;
+
+        public PollingWildCardChangeToken(
+            string root,
+            string pattern)
+            : this(
+                new DirectoryInfoWrapper(new DirectoryInfo(root)),
+                pattern,
+                Internal.Clock.Instance)
+        {
+        }
+
+        public PollingWildCardChangeToken(
+            DirectoryInfoBase directoryInfo,
+            string pattern,
+            IClock clock)
+        {
+            _directoryInfo = directoryInfo;
+            Clock = clock;
+
+            _matcher = new Matcher(StringComparison.OrdinalIgnoreCase);
+            _matcher.AddInclude(pattern);
+            CalculateChanges();
+        }
+
+        /// <inheritdoc />
+        public bool ActiveChangeCallbacks => false;
+
+        public IClock Clock { get; }
```

Comment:
We could consider commonizing this with the one in Caching and Kestrel - https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/ISystemClock.cs. Security has this type too but it's public API: https://github.com/aspnet/Security/blob/22d133cefa20926b25eeb82779343b23a221117f/src/Microsoft.AspNetCore.Authentication/ISystemClock.cs